pkgs/by-name/md/mdbook-pandoc/package.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pandoc
+, texliveSmall
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "mdbook-pandoc";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "max-heller";
+    repo = "mdbook-pandoc";
+    rev= "v${version}";
+    hash = "sha256-lsgvpS7OoIwANERWnPcBEqYvZAKKsrPoufJOXbXxOkg=";
+  };
+
+  cargoHash = "sha256-oBS2QWQJM4QrYJz37+mldAxI5XOVYMuOKqNiRgAGICs=";
+
+  postPatch = ''
+    for pandocUser in src/{pandoc.rs,pandoc/renderer.rs}; do
+      substituteInPlace $pandocUser \
+        --replace-fail 'Command::new("pandoc")' 'Command::new("${pandoc}/bin/pandoc")'
+    done
+  '';
```

Comment:
This is not necessary, please remove it.

Reply:
I've added this, because mdbook-pandoc depends on pandoc being available at runtime. The tests also depend on pandoc being available.

I can of course remove this and add pandoc to `nativeCheckInputs`. Though that has the consequence, that a user installing only mdbook-pandoc, but not pandoc itself, would end up with a 'broken' mdbook-pandoc. 

This is the first time I'm packaging a program depending on other executables being in PATH at runtime, so this might also just be my expectations on how to do this being wrong.

Reply:
As far as I know, you can add `pandoc` to `propagatedBuildInputs` to achieve what you want.

Reply:
Now that you mention `propagatedBuildInputs`, I remember that I in fact already did something similar in https://github.com/NixOS/nixpkgs/pull/261216#pullrequestreview-1682783107.

Over there, the conclusion was, that `propagatedBuildInputs` does not add the dependencies to PATH, and in the end I/we ended up with something similar to the diff above.

Reply:
i believe `wrapProgram` is the typical way to solve this without manually patching source code.

the other advantage of `wrapProgram` is it will also allow suffixing to PATH, which is desirable so that if the user installed a different version of pandoc (eg. via their system package manager), that version will be used instead of the one in nixpkgs.